TestUnitConcurrentGetNodeReadmit can panic locally on Windows

## Summary

`TestUnitConcurrentGetNodeReadmit` can panic locally on Windows with:

```text
panic: failed to find a healthy working node
```

This was observed while validating #1708, but it also reproduces on a clean `origin/main` checkout, so it does not appear to be introduced by that PR.

## Environment

```text
go version go1.25.7 windows/amd64
```

Baseline reproduced on clean `origin/main` at:

```text
d91f3f2b1938fcd651b43bff9eeed7ca33fe5ec1
```

## Reproduction

```bash
go test ./sdk -tags="unit" -v -run TestUnitConcurrentGetNodeReadmit -count=10 -timeout 9999s
```

## Observed

The test can panic with:

```text
panic: failed to find a healthy working node
sdk/managed_network.go:185
sdk/network_unit_test.go:85
```

## Notes

The PR-specific test from #1708 passes locally:

```bash
go test ./sdk -tags="unit" -v -run TestUnitHbarAllowance -timeout 9999s
```

The GitHub Actions `Unit and Integration Tests (unit)` job is also passing on #1708, so this may be environment-specific or timing-sensitive.

--- a/sdk/managed_network.go
+++ b/sdk/managed_network.go
@@ -100,7 +100,11 @@ func (mn *_ManagedNetwork) _ReadmitNodes() {
 	mn.healthyNodesMutex.Lock()
 	defer mn.healthyNodesMutex.Unlock()
 
-	if mn.earliestReadmitTime.Before(now) {
+	mn._ReadmitNodesLocked(now)
+}
+
+func (mn *_ManagedNetwork) _ReadmitNodesLocked(now time.Time) {
+	if !mn.earliestReadmitTime.After(now) {
 		nextEarliestReadmitTime := now.Add(mn.maxNodeReadmitPeriod)
 
 		for _, node := range mn.nodes {
@@ -122,7 +126,7 @@ func (mn *_ManagedNetwork) _ReadmitNodes() {
 				}
 			}
 
-			if node._GetReadmitTime().Before(now) {
+			if !node._GetReadmitTime().After(now) {
 				mn.healthyNodes = append(mn.healthyNodes, node)
 			}
 		}
@@ -177,17 +181,20 @@ func (mn *_ManagedNetwork) _SetMinBackoff(minBackoff time.Duration) {
 }
 
 func (mn *_ManagedNetwork) _GetNode() _IManagedNode {
-	mn._ReadmitNodes()
-	mn.healthyNodesMutex.RLock()
-	defer mn.healthyNodesMutex.RUnlock()
+	mn.healthyNodesMutex.Lock()
+	mn._ReadmitNodesLocked(time.Now())
 
 	if len(mn.healthyNodes) == 0 {
+		mn.healthyNodesMutex.Unlock()
 		panic("failed to find a healthy working node")
 	}
 
-	bg := big.NewInt(int64(len(mn.healthyNodes)))
+	healthyNodes := slices.Clone(mn.healthyNodes)
+	mn.healthyNodesMutex.Unlock()
+
+	bg := big.NewInt(int64(len(healthyNodes)))
 	index, _ := rand.Int(rand.Reader, bg)
-	return mn.healthyNodes[index.Int64()]
+	return healthyNodes[index.Int64()]
 }
 
 func (mn *_ManagedNetwork) _GetMinBackoff() time.Duration {

